src/omotes_simulator_core/entities/assets/controller/controller_producer.py

```diff
@@ -38,10 +39,12 @@ def __init__(
         :param float temperature_supply: Supply temperature of the source.
         :param float temperature_return: Return temperature of the source.
         :param float power: Power of the source.
+        :param float marginal_costs: Marginal costs of the source.
         :param int priority: Priority of the source.
         """
         super().__init__(name, identifier)
         self.temperature_return: float = temperature_return
```

Comment:
This type defintion seems to be 'double'? Can you quickly fix it?

Reply:
nope, it is float, the method get_marginal_costs passes back a float,
`        return float(self.esdl_asset.costInformation.marginalCosts.value)`

Reply:
My bad, it is a misunderstanding. I fell over the fact that you define the variable that goes into the object as float and thereafter set the type of the self.marginal_costs also to float.